pkg/prometheus/promcfg.go

```diff
@@ -1131,7 +1135,14 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 	}
 
 	selectedNamespaces := getNamespacesFromNamespaceSelector(&m.Spec.NamespaceSelector, m.Namespace, ignoreNamespaceSelectors)
-	cfg = append(cfg, cg.generateK8SSDConfig(selectedNamespaces, apiserverConfig, store, kubernetesSDRoleEndpoint))
+
+	role := kubernetesSDRoleEndpoint
+	if cg.version.GTE(semver.MustParse("2.21.0")) {
+		if isSupportEndpointSliceResource {
```

Comment:
```suggestion
	if cg.version.GTE(semver.MustParse("2.21.0")) && isSupportEndpointSliceResource {
```
Could we inline the two conditions to avoid nesting?